File: android/app/src/main/java/com/bazical/app/ui/feedback/FeedbackDetailViewModel.kt
package com.bazical.app.ui.feedback

import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.bazical.app.domain.repository.FeedbackRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class FeedbackDetailViewModel @Inject constructor(
    private val feedbackRepository: FeedbackRepository,
    savedStateHandle: SavedStateHandle
) : ViewModel() {

    private val feedbackId: String = savedStateHandle.get<String>("id") ?: ""

    private val _uiState = MutableStateFlow(FeedbackDetailUiState())
    val uiState: StateFlow<FeedbackDetailUiState> = _uiState.asStateFlow()

    init {
        loadFeedbackDetail()
    }

    fun loadFeedbackDetail() {
        viewModelScope.launch {
            _uiState.update { it.copy(isLoading = true, errorMessage = null) }
            feedbackRepository.getFeedbackDetail(feedbackId)
                .onSuccess { feedback ->
                    _uiState.update {
                        it.copy(
                            feedback = feedback,
                            replies = feedback.replies,
                            isLoading = false
                        )
                    }
                }
                .onFailure { error ->
                    _uiState.update {
                        it.copy(
                            isLoading = false,
                            errorMessage = error.message ?: "加载失败"
                        )
                    }
                }
        }
    }

    fun updateReplyContent(content: String) {
        _uiState.update { it.copy(replyContent = content) }
    }

    fun sendReply() {
        val content = _uiState.value.replyContent.trim()
        if (content.isEmpty()) return

        viewModelScope.launch {
            _uiState.update { it.copy(isSending = true, errorMessage = null) }
            feedbackRepository.addReply(feedbackId, content)
                .onSuccess {
                    _uiState.update { it.copy(isSending = false, replyContent = "") }
                    loadFeedbackDetail()
                }
                .onFailure { error ->
                    _uiState.update {
                        it.copy(
                            isSending = false,
                            errorMessage = error.message ?: "发送失败"
                        )
                    }
                }
        }
    }

    fun clearError() {
        _uiState.update { it.copy(errorMessage = null) }
    }
}

File: ios/BaziCalApp/Features/Feedback/FeedbackDetailViewModel.swift
import Foundation
import SwiftUI

/// 反馈详情 UI 状态
struct FeedbackDetailUiState {
    var feedback: Feedback?
    var replies: [Reply] = []
    var isLoading: Bool = false
    var errorMessage: String?
    var replyContent: String = ""
    var isSending: Bool = false
}

/// 反馈详情视图模型
@MainActor
class FeedbackDetailViewModel: ObservableObject {
    @Published var uiState = FeedbackDetailUiState()

    private let apiService = APIService.shared
    private let feedbackId: String

    init(feedbackId: String) {
        self.feedbackId = feedbackId
        Task {
            await loadFeedbackDetail()
        }
    }

    /// 加载反馈详情
    func loadFeedbackDetail() async {
        uiState.isLoading = true
        uiState.errorMessage = nil

        do {
            uiState.feedback = try await apiService.getFeedbackDetail(id: feedbackId)
            uiState.replies = uiState.feedback?.replies ?? []
        } catch {
            uiState.errorMessage = error.localizedDescription
        }

        uiState.isLoading = false
    }

    /// 更新回复内容
    func updateReplyContent(_ content: String) {
        uiState.replyContent = content
    }

    /// 发送回复
    func sendReply() async {
        guard !uiState.replyContent.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            return
        }

        uiState.isSending = true

        do {
            let reply = try await apiService.sendFeedbackReply(
                feedbackId: feedbackId,
                content: uiState.replyContent
            )
            uiState.replies.append(reply)
            uiState.replyContent = ""
        } catch {
            uiState.errorMessage = error.localizedDescription
        }

        uiState.isSending = false
    }
}